### PR TITLE
Fix pretty printing tests on Windows.

### PR DESCRIPTION
**Pull request by wilhuff**

### CODE REVIEW

File: Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc [hunk 1]
```diff
@@ -44,8 +44,16 @@ using ::testing::MatchesRegex;
...
+  std::string hex = "0x[0-9A-Fa-f]+";
+#else
+  // On windows, validate more loosely because GoogleTest doesn't support
+  // matching with brackets.
+  std::string hex = "0x\\w+";
+#endif
+
   // NOLINTNEXTLINE(whitespace/braces)
-  std::string header = std::string{"<"} + proto_name + " 0x[0-9A-Fa-f]+>: \\{";
+  std::string header = std::string{"<"} + proto_name + " " + hex + ">: \\{";
```
Comment (var-const) [inline Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc]: Optional: the split between `hex` and string literals here isn't obvious. Consider either moving the single space into `hex` or moving `0x` into the string literal:
```cpp
+ proto_name + " 0x" + hex + ...
// or
+ proto_name + hex + ...
```

File: Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc [hunk 2]
```diff
@@ -44,8 +44,16 @@ using ::testing::MatchesRegex;
 // contain any newlines; the appropriate newlines must be included in
 // `contents`.
 MATCHER_P2(IsPrintedAs, proto_name, contents, "") {
+#if GTEST_USES_POSIX_RE
+  std::string hex = "0x[0-9A-Fa-f]+";
+#else
+  // On windows, validate more loosely because GoogleTest doesn't support
```
Comment (var-const) [inline Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc]: Nit: capitalize `Windows`.

Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc]: Done.

File: Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc (hunk 1 above)
Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc]: Done. Also changed to use `absl::StrCat`.

Comment (var-const) [reply Firestore/core/test/firebase/firestore/nanopb/pretty_printing_test.cc]: Great, thanks! I still forget about `StrCat` from time to time.

**Pull request merged**